### PR TITLE
Add test feature flags

### PR DESCRIPTION
**Pull request by rjshade**
Adds test feature flags to allow fixtures to selectively disable the following tests:
- resource_quota_server
- network_status_change

Adds some comments to describe use of test feature flags.

### CODE REVIEW

File: test/core/end2end/end2end_tests.h [hunk 1]
```diff
@@ -39,12 +39,17 @@
...
+/* Test feature flags. If FEATURE_MASK_SUPPORTS_X is included in the
+ * feature_mask of a grpc_end2end_test_config then tests of feature X will be
+ * disabled for that test fixture. */
 #define FEATURE_MASK_SUPPORTS_DELAYED_CONNECTION 1
 #define FEATURE_MASK_SUPPORTS_HOSTNAME_VERIFICATION 2
 #define FEATURE_MASK_SUPPORTS_PER_CALL_CREDENTIALS 4
 #define FEATURE_MASK_SUPPORTS_REQUEST_PROXYING 8
 #define FEATURE_MASK_SUPPORTS_CLIENT_CHANNEL 16
 #define FEATURE_MASK_SUPPORTS_AUTHORITY_HEADER 32
+#define FEATURE_MASK_SUPPORTS_RESOURCE_QUOTA_SERVER 64
```
Comment (yang-g) [inline test/core/end2end/end2end_tests.h]: I think you want something like FEATURE_MASK_FORBIDS_XXX?
I think some if not all other flags works the other way and for easier integration with existing configs, you may want something to default to 0 (support) and set to 1 in places where it is not supported.

**Comment by rjshade** [comment]
Renamed flags to explicitly say that they are for disabling features (DOES_NOT_SUPPORT)

**Comment by rjshade** [comment]
@yang-g @muxi any idea why "Basic Tests Mac" is still pending? It's been in that state for hours now.

**Comment by muxi** [comment]
I am troubleshooting. Appears some test is taking forever to run.

**Comment by rjshade** [comment]
@muxi Thanks, how do you get from here to the Jenkins status page showing details of the pending test run? There's no "Details" link.

**Comment by muxi** [comment]
There's no such info. I did a manual run of the tests with verbose information and fount that. The fix is in place. Let's see how it goes.

**Comment by rjshade** [comment]
@muxi Can I merge this pull request?

**Pull request merged**